### PR TITLE
[WIP] Hopefully working example

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

 This is not meant to be merged but to help provide implementation details on how to export parthenon. 

## PR Checklist

- [ ] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -188,12 +194,34 @@ endif()
...
 else()
   find_package(Kokkos 3 REQUIRED)
 endif()
 
+#add_library(Kokkos SHARED IMPORTED)
+#add_library(Kokkos INTERFACE)
+#target_link_libraries(Kokkos INTERFACE ${Kokkos_LIBRARIES})
+#target_compile_definitions(Kokkos INTERFACE ${Kokkos_DEFINITIONS})
+#target_include_directories(Kokkos INTERFACE ${Kokkos_INCLUDE_DIRS})
+#target_sources(Kokkos INTERFACE ${Kokkos_SOURCES})
```
Comment (JoshuaSBrown) [inline CMakeLists.txt:215]: @brryan you can also treat Kokkos as interface library similar to HDF5_C, I just don't know the consequences of doing this, I think I also got your sample working doing this. You need to add these lines and the install target command on line 219 below I think.

**Comment by Yurlungur** [comment]
@JoshuaSBrown if this ends up working, can you add a doc explaining what you did and merge that in?

**Comment by brryan** [comment]
@JoshuaSBrown Getting closer to being able to configure/build a standalone calculate_pi with cmake with this! I did change `DESTINATION lib64/cmake/parthenon` to `DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/parthenon"` Then, with the `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.15)                                             
                                                                                 
project(parthenon_linking_example)                                               
                                                                                 
set(PARTHENON_PATH                                                               
    "/home/brryan/builds/parthenon/install-brryan/install_target/lib/cmake/parthenon")
set(KOKKOS_PATH                                                                  
    "/home/brryan/builds/parthenon/install-brryan/install_target/lib/cmake/Kokkos")
set(CMAKE_PREFIX_PATH "${CMAKE_PREFIX_PATH}" "${PARTHENON_PATH}" "${KOKKOS_PATH}")
find_package(parthenon REQUIRED)                                                 
                                                                                 
add_executable( pi-example                                                       
  calculate_pi.cpp                                                               
  calculate_pi.hpp                                                               
  pi_driver.cpp                                                                  
  pi_driver.hpp                                                                  
)                                                                                
target_link_libraries(pi-example PRIVATE parthenon)                              
target_include_directories(pi-example PRIVATE                                    
  "/home/brryan/builds/parthenon/install-brryan/install_target/include"          
  "/home/brryan/builds/parthenon/install-brryan/install_target/include/kokkos")
```
I can configure, but linking fails:
```
(py3) brryan@ubuntu:~/Documents/parth_link/build$ cmake ../
-- Enabled Kokkos devices: SERIAL
-- Configuring done
-- Generating done
-- Build files have been written to: /home/brryan/Documents/parth_link/build
(py3) brryan@ubuntu:~/Documents/parth_link/build$ make
Scanning dependencies of target pi-example
[ 33%] Building CXX object CMakeFiles/pi-example.dir/calculate_pi.cpp.o
[ 66%] Building CXX object CMakeFiles/pi-example.dir/pi_driver.cpp.o
[100%] Linking CXX executable pi-example
/usr/bin/ld: cannot find -lparthenon
collect2: error: ld returned 1 exit status
CMakeFiles/pi-example.dir/build.make:118: recipe for target 'pi-example' failed
make[2]: *** [pi-example] Error 1
CMakeFiles/Makefile2:95: recipe for target 'CMakeFiles/pi-example.dir/all' failed
make[1]: *** [CMakeFiles/pi-example.dir/all] Error 2
Makefile:103: recipe for target 'all' failed
make: *** [all] Error 2
```

Were you able to link to `libparthenon.so` with cmake, and if so could I see your `CMakeLists.txt`? Also, I shouldn't need to manually specify the include directories for parthenon/Kokkos headers using `target_include_directories` in the above once I have a cmake target, right?

**Comment by JoshuaSBrown** [comment]
> Were you able to link to libparthenon.so with cmake, and if so could I see your CMakeLists.txt? Also, I shouldn't need to manually specify the include directories for parthenon/Kokkos headers using target_include_directories in the above once I have a cmake target, right?

I didn't get that far, I can take a look again on Monday (just push your latest changes by then if you want me to take a look), hey we are getting closer :).

**Comment by JoshuaSBrown** [comment]
@brryan I'm going to delete this unless you are still using it as are reference?

**Comment by brryan** [comment]
@JoshuaSBrown I'm all set, feel free to delete

**Pull request closed without merging**